### DISCUSSION TITLE
Patroni db mode uses source database "template1"  [ URGENT ]

### DISCUSSION THREAD

**Discussion by muhabbad** [cybertec-postgresql/pgwatch2#614] [Q&A] [answered]
I use Patroni db type in my instances.yaml file. And my pgwatch2 collector uses template1 database. In documentation, it is stated that :

> NB! All “continuous” modes expect access to “template1” or “postgres” databasess of the specified cluster to determine the database names residing there.

In my case, i don't use continuous mode in my configuration. What should i do? Is this expected behavior in Patroni mode?

**Comment by muhabbad** [comment]
This is urgent for me. Is there someone to help me?

**Comment by pashagolub** [comment, marked as answer]
Of course, please check support options here: https://www.cybertec-postgresql.com/en/services/postgresql-support/standard/product-support/

**Comment by markwort** [comment]
When using the "patroni" mode, pgwatch2 finds the database instances (cluster members) to monitor via the information that patroni publishes for each "member" to etcd for example.
That information (the `conn_url` key in json data for every member) contains the database name to use when connecting: `postgres` . That is hard coded in patroni.

Do you need to use the patroni mode so that pgwatch2 finds all of your patroni cluster members? Then you need to enable pgwatch2 to connect to the or postgres database.

If you want to use a non-dynamic configuration (i.e. you specify the connection details to each cluster member yourself), you can use the regular "postgres" mode, it doesn't matter to pgwatch2 if there is patroni managing those instances or not.
In that case you can specify the `dbname` directly in `instances.yaml` and pgwatch2 will use that for connections.

If you encountered an issue, it would be useful to include the exact version of pgwatch2, configuration and relevant error messages that you saw.
